### DISCUSSION TITLE
FQDN resolution to v6 as remote_addrs in swanctl.conf

### DISCUSSION THREAD

**Discussion by kaizoku-o** [strongswan/strongswan#1006] [Q&A] [answered]
Hi,

I had a question related to FQDN resolution in strongSwan:

How do I know if an FQDN set as remote_addrs in swanctl.conf is resolved as v4 or v6 address? Is there a config option to set either v4 or v6 as preferred? 

(Apologies if this was in the documentation and I missed it.)

Also another question related to FQDN but setting it in remote_id:
From the documentation:
"If the string begins with @, the type is set to FQDN and the encoding is the literal string after that prefix. In strongSwan versions before 5.0.0 this prefix prevented that a FQDN was resolved into an IP address whereas current versions don’t automatically resolve FQDNs when parsing identities."
Is there a way to ask strongSwan to resolve FQDN in remote_addrs as well, if it doesn't do it automatically?

**Comment by tobiasbrunner** [comment, marked as answer]
> How do I know if an FQDN set as remote_addrs in swanctl.conf is resolved as v4 or v6 address? Is there a config option to set either v4 or v6 as preferred?

Depends on the behavior of the DNS resolver on your system. But most probably prefer IPv6 over IPv4. You can disable address families completely via the [strongswan.conf options](https://docs.strongswan.org/strongswan-docs/5.9/config/strongswanConf.html#_charon_plugins_socket_default) for the _socket-default_ plugin.

> Is there a way to ask strongSwan to resolve FQDN in remote_addrs as well, if it doesn't do it automatically?

They are always resolved automatically. Only identities are never resolved.

**Reply by kaizoku-o** [reply]
Thanks for the answer! Oops sorry I meant to ask is there a way to ask strongSwan to resolve "remote_id" if it is set to FQDN? (I guess the answer is no, since identities are never resolved?)

**Reply by kaizoku-o** [reply]
Also, sorry to pile on to this question but what is the FQDN resolution behavor if the public ip changes?

**Reply by tobiasbrunner** [reply]
>  (I guess the answer is no, since identities are never resolved?)

Correct.

> Also, sorry to pile on to this question but what is the FQDN resolution behavor if the public ip changes?

They are resolved on-demand not when the connection is loaded if that answers your question.

**Reply by kaizoku-o** [reply]
Thanks!
